packages/source-commonmark/src/index.ts

```diff
@@ -1,9 +1,197 @@
-import HTMLSource from '@atjson/source-html';
+import Document, { Annotation } from '@atjson/document';
+import * as entities from 'entities';
 import * as MarkdownIt from 'markdown-it';
+import schema from './schema';
 
-export default class extends HTMLSource {
+export { default as schema } from './schema';
+
+interface Attributes {
+  [key: string]: string;
+}
+
+type Tuple = [string, string];
+
+function getAttributes(token: MarkdownIt.Token): Attributes {
+  return (token.attrs || []).reduce((attributes: Attributes, attribute: Tuple) => {
+    attributes[attribute[0]] = attribute[1];
+    return attributes;
+  }, {});
+}
```

Comment:
ok - pausing again - so far so good!

Reply:
Thanks for the really thorough review, @pgoldrbx! Great to have your eyes on all this stuff, too.